TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -391,5 +391,43 @@ public LockTableStatement CreateLockTableStatement(CodeElementsParser.LockTableS
             var savepointName = context.Diagnostics == null ? new SymbolReference(new AlphanumericValue((Token)context.savepoint_name), SymbolType.SqlIdentifier) : null;
             return new ReleaseSavepointStatement(savepointName);
         }
+
+        public ExecuteImmediateStatement CreateExecuteImmediateStatement(CodeElementsParser.ExecuteImmediateStatementContext context)
+        {
+            SqlExpression sqlExpression = null;
+            if (context.sqlExpression() != null)
+            {
+                sqlExpression = CreateSqlExpression(context.sqlExpression());
+            }
+
+            //TODO add support for string expressions
+            return new ExecuteImmediateStatement(sqlExpression);
+        }
+
+        private SqlVariable CreateSqlVariable(CodeElementsParser.SqlVariableContext context)
+        {
+            if (context.hostVariable() != null)
+            {
+                return CreateSqlHostVariable(context.hostVariable());
+            }
+            //TODO Add other conditions when adding new  Sql Variable Types 
+            return null;
+        }
+        private SqlExpression CreateSqlExpression(CodeElementsParser.SqlExpressionContext context)
+        {
+            if (context.column_name() != null)
+            {
+                return CreateSqlColumnName(context.column_name());
+            }
+            else if (context.sqlConstant() != null)
+            {
+                return new SqlConstant(ParseTreeUtils.GetFirstToken(context.sqlConstant()));
```

Comment:
Beware, you forgot about datetime_constant, you need to create a new `CreateSqlConstant` method which will handle various types of sql constants